### PR TITLE
Tests for classpath approval

### PR DESCRIPTION
**Pull request by olivergondza**
https://github.com/jenkinsci/script-security-plugin/pull/125

### CODE REVIEW

**Comment by ikedam** [comment]
-1
I cannot get what you want.
Do you really want to merge this request to groovy-postbuild?
Why don't you add tests to script-security itself?

**Comment by olivergondza** [comment]
I can be tested in script-security though that would not ensure the message reported by this plugin is not misleading...

**Comment by ikedam** [comment]
I cannot get what you want yet.
I don't think this test is necessary for groovy-postbuild.
What groovy-postbuild have to do is to output the exception from the script-security, and not to ensure what the message is. It's an issue of script-security.
It's really bad idea to make tests depending on the message strings of other plugins as it can be easily broken when the message is modified or localized in a future version.

**Comment by olivergondza** [comment]
@ikedam, fair enough. Though, the plugins is missing classpath tests completely so I will rework the tests not to depend on script-security so much and reopen.

File: pom.xml [hunk 1]
```diff
@@ -75,7 +75,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>script-security</artifactId>
-            <version>1.15</version>
+            <version>1.19</version> <!-- JENKINS-24399 -->
```
Comment (ikedam) [inline pom.xml]: I don't think this is necessary as we don't have to test JENKINS-24399 here.
It's already tested in script-security.
I prefer preserving compatibility (or requirements in this case).

**Comment by ikedam** [comment]
Please update the subject of this request, as it's difficult to get what is done when left in the git log.

File: pom.xml (hunk 1 above)
Comment (olivergondza) [inline pom.xml]: The thing is the directory class path would be proposed for approval and can in fact be approved with all the security implications JENKINS-24399 have. This is needed to keep the tests passing and the further secure the plugin.

**Comment by ikedam** [comment]
I never think this test makes sense. And I never think it's necessary to make change requirements of this plugin for that.
Maintenance of plugins should be isolated.

Even if this test fails (and other tests passed), it means not a problem in this plugin, but only a problem in script-security. There's nothing this plugin can do for that, and that means this test is unnecessary for this plugin.
Too few tests are not good, of course, but too many tests are not good, too.

**Comment by ikedam** [comment]
I think it makes sense to test approving non-whitelisted methods and it also ensures this plugin is correctly integrated with script-security.
I'll consider to add that test.

**Pull request closed without merging**